### ISSUE TITLE
feat: Pre-commit hooks for linting and tests

### ISSUE THREAD

**Issue by JaeAeich** [elixir-cloud-aai/cloud-components#35]
**Is your feature request related to a problem? Please describe.**

Currently, there is a lack of testing and linting enforcement before committing code, which can result in unsupervised code being pushed and have detrimental effects on code quality and the code review process.

**Describe the solution you'd like**

I propose adding a pre-commit hook that includes running test scripts and performing linting checks on the codebase. To optimize the linting process for the growing codebase, we can leverage the 'lint-staged' npm package. 'lint-staged' enables us to selectively lint and fix only the files that are staged for commit, reducing potential delays caused by the entire codebase's linting. By implementing this pre-commit hook, we ensure that testing and linting are enforced before code is committed, leading to improved code quality and a more effective code review process

**Acknowledgement**

> We thank you for your valuable contribution and would like to acknowledge it
> on our [contributors][contributors] page. To help us with that, we would like
> to kindly request that you help us with this by commenting on this issue with
> a request to the [@all-contributors bot][all-contributors-bot] of the form
> `@all-contributors please add @YOUR_GH_HANDLE for TYPE_1, TYPE_2, ...`, where
> `TYPE_1` etc. refer to [contribution types supported by the All Contributors
> Specification][all-contributors-types], e.g., `idea`.

[all-contributors-bot]: <https://allcontributors.org/docs/en/bot/overview>
[all-contributors-types]: <https://allcontributors.org/docs/en/emoji-key>
[contributors]: ../../CONTRIBUTORS

**Comment by JaeAeich** [comment]
@git-anurag-hub Since Turborepo already has a caching feature, it doesn't matter if we run lint command for all the files or not, in that case, staged-lint would be a redundant layer imo. Is that right? if so a simple `turbo lint` and `turbo test` scripts in the pre hook should do the job or does the staged-lint path seems more appropriate ?

**Comment by uniqueg** [comment]
@JaeAeich: Well, if you are not a regular developer and you only check out the repo for a small fix, you'd have to run the linting for _all_ packages. Still, I tend to agree. If the repo ever gets so big that it becomes a problem, we can still change that. So probably best to keep it agile here :)

**Comment by git-anurag-hub** [comment]
Yeah agree with @uniqueg, you can just run `npm run lint` & `npm run test` as the pre-commit hook & it will do the thing ig!

**Comment by JaeAeich** [comment]
@uniqueg @git-anurag-hub Just a question do we assume that the developer has already cached a dev instance by running `npm run dev` before committing any changes otherwise, since the dependencies need to be active when running the `lint` and `test` script otherwise the process will fail. If so, this would just be a matter of adding the `lint` and `test` script to the pre-commit hook else we might need other dependencies.

**Comment by uniqueg** [comment]
From my PoV: As long as such assumptions are documented, I think that's totally valid.

**Comment by JaeAeich** [comment]
In that case, I'll update the concerning REAMDE and create the PR.
